sdk/go/common/apitype/core.go

```diff
@@ -329,6 +329,8 @@ type ResourceV3 struct {
 	Parent resource.URN `json:"parent,omitempty" yaml:"parent,omitempty"`
 	// Protect is set to true when this resource is "protected" and may not be deleted.
 	Protect bool `json:"protect,omitempty" yaml:"protect,omitempty"`
+	// Taint is set to true when we wish to force it to be replaced upon the next update.
+	Taint bool `json:"taint,omitempty" yaml:"taint,omitempty"`
```

Comment:
This should probably be added in the [`supportedFeatures`](https://github.com/pulumi/pulumi/blob/master/pkg/resource/stack/deployment.go#L112) list.  Right now if we set this with a newer version of pulumi, and then run an operation on the stack with an older version, the taint flag is ignored, and worse (I think) removed from the state.  Adding `taint` to the `supportedFeatures` list will make sure older CLI versions will error out if they encounter this flag.

Reply:
Did not realize we had this! Is there more documentation about its intended usage and when it is required? The comment was not sufficiently enlightening for me to tell: "supportedFeatures is a map of features that are currently supported. Any features not in this map will be rejected."